CIScripts/Tests/VRouterAgentTests.ps1

```diff
@@ -255,6 +260,108 @@ function Test-VRouterAgentIntegration {
+            $LASTEXITCODE
+            $DockerOutput
+        }
+        $ExitCode = $Output[0]
+        $Message = $Output[1]
+        if ($ExitCode -ne 0) {
+            Write-Host "    Exit code: $ExitCode"
+            Write-Host "    Docker output: $Message"
+        }
+        return $ExitCode
+    }
+
+    function Assert-PingSucceeded {
+        Param ([Parameter(Mandatory = $true)] [Object[]] $Output)
+        $ErrorMessage = "Ping failed. EXPECTED: Ping succeeded."
+        Foreach ($Line in $Output) {
+            if ($Line -match ", Received = (?<NumOfReceivedPackets>[\d]+),[.]*") {
+                if ($matches.NumOfReceivedPackets -gt 0) {
+                    return
+                } else {
+                    throw $ErrorMessage
+                }
+            }
+        }
+        throw $ErrorMessage
+    }
+
+    function Test-Ping {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session1,
+               [Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session2,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration,
+               [Parameter(Mandatory = $true)] [string] $Container1Name,
+               [Parameter(Mandatory = $true)] [string] $Container2Name)
+        Write-Host "======> Given Docker Driver and Extension are running"
+
+        # 1st compute node
+        Clear-TestConfiguration -Session $Session1 -TestConfiguration $TestConfiguration
+        Initialize-TestConfiguration -Session $Session1 -TestConfiguration $TestConfiguration
+        Assert-ExtensionIsRunning -Session $Session1 -TestConfiguration $TestConfiguration
+
+        # 2nd compute node (if there actually is more than 1 compute node)
+        if ($Session1 -ne $Session2) {
+            Clear-TestConfiguration -Session $Session2 -TestConfiguration $TestConfiguration
+            Initialize-TestConfiguration -Session $Session2 -TestConfiguration $TestConfiguration
+            Assert-ExtensionIsRunning -Session $Session2 -TestConfiguration $TestConfiguration
+        }
+
+        Write-Host "======> Given Agent is running"
+
+        # 1st compute node
+        New-AgentConfigFile -Session $Session1 -TestConfiguration $TestConfiguration
+        Enable-VRouterAgent -Session $Session1 -ConfigFilePath $TestConfiguration.AgentConfigFilePath
+        Assert-AgentIsRunning -Session $Session1
+
+        # 2nd compute node (if there actually is more than 1 compute node)
+        if ($Session1 -ne $Session2) {
+           New-AgentConfigFile -Session $Session2 -TestConfiguration $TestConfiguration
+           Enable-VRouterAgent -Session $Session2 -ConfigFilePath $TestConfiguration.AgentConfigFilePath
+           Assert-AgentIsRunning -Session $Session2
+        }
+
+        Start-Sleep -Seconds 15
+
+        Write-Host "======> Given 2 containers belonging to the same network are running"
```

Comment:
I think we should structure those tests this way:

```
Given Docker Driver and Extension are running
Given Agent is running
When 2 containers belonging to the same network are running
Then ping between them succeeds
```

This way we could encapsulate ping command execution and checking its output in one function.